crates/contracts/src/world/world_contract.cairo

```diff
@@ -819,11 +977,19 @@ pub mod world {
         ///
         /// * `keys` - The keys of the event.
         /// * `values` - The data to be logged by the event.
-        fn emit(self: @ContractState, mut keys: Array<felt252>, values: Span<felt252>) {
-            let system = get_caller_address();
-            system.serialize(ref keys);
-
-            emit_event_syscall(keys.span(), values).unwrap_syscall();
+        fn emit(
+            ref self: ContractState,
+            event_selector: felt252,
+            keys: Span<felt252>,
+            values: Span<felt252>,
+            historical: bool
+        ) {
+            EventEmitter::emit(
```

Comment:
We don't have permissions check here. It is fair to say that you can only emit an event if you have write access to it? @Larkooo @remybar ?

Reply:
Is there a real safety issue to emit an unexpected event from a "malicious" contract ?
I would say that, as the address of the system which emitted the event is stored in the event data, it's easy to filter out malicious events.
On the other hand, it can be heavy for a project to handle permissions for all its events.

Reply:
We checked with @Larkooo and it seems we need permissions. The point I feel hard it's the cost.. since permissions checks is expensive. But without permissions, everybody could overwrite some values... which could lead to unexpected client behavior.